javascript/javascript1/week4/homework/voice-assistant.js

```diff
@@ -0,0 +1,129 @@
+let userData = {
+  name: null,
+  todos: [],
+};
+
+const processNameCommand = (command) => {
+  const name = command.replace('Hello my name is', '').trim();
+  if (name.length > 0) {
+    userData.name = name;
+    return `nice to meet you ${name}`;
+  }
+  return 'Invalid input of name.';
+};
+
+const processAddToDoCommand = (command) => {
+  let replacedAdd = command.replace('Add', '').trim();
+  const toDo = replacedAdd.replace('to my todo', '').trim();
+  if (toDo.length > 0) {
+    userData.todos.push(toDo);
+    return `${toDo} added to your todo`;
+  }
+  return 'Invalid input of to do list.';
+};
+
+const processRemoveToDoCommand = (command) => {
+  let replacedRemove = command.replace('Remove', '').trim();
+  const toDo = replacedRemove.replace('from my todo', '').trim();
+  if (toDo.length> 0) {
+    userData.todos = userData.todos.filter((todo) => {
+      return todo !== toDo;  
+  });
+  return `Removed ${toDo} from your todo`;
+  }  
+   return 'Invalid input of to do list.'
+};
+
+const setATimer = (command) => {
+  let replacedSet = command.replace('Set a timer for', '').trim();
+  let minutesString = replacedSet.replace('minutes', '').trim();
+  const minutes = parseFloat(minutesString)
+  setTimeout(() => console.log("Timer done"), minutes * 60 * 1000);
+  return `Timer is set for ${minutes} minutes`
+}
+
+const doMath = (command) => {
+  let replacedWhatIs = command.replace('What is', '').trim();
+  const symbol = replacedWhatIs.split(' ')[1];
+  const number1 = parseInt(replacedWhatIs.split(symbol)[0]);
+  const number2 = parseInt(replacedWhatIs.split(symbol)[1]);
+  const symbolsArray = ['+', '-', '*', '/', '%']
+    const isAllowedSymbol = symbolsArray.some((allowedSymbol) => {
+      return allowedSymbol === symbol
+    })
+  if (!isNaN(number1) && !isNaN(number2) && isAllowedSymbol === true) {
```

Comment:
Since `.some()` returns true or false, You can avoid writing `isAllowedSymbol === true`. `if (isAllowedSymbol)` is same as `if(isAllowedSymbol === true)` .  `if( !isAllowedSymbol )` is the same with `if(isAllowedSymbol === false)`.